course-v2/layouts/_default/baseof.html

```diff
@@ -176,6 +176,17 @@
     }
   </script>
 
+  <script>
```

Comment:
@ibrahimjaved12 glad you picked this up. Will be a really big improvement for OCW.

What would you think of using a dynamic `import()` for this? Something like:
```ts
// course-v2/assets/videojs-imports.ts
// exactly as you have it now

// course-v2/assets/some_other_file.ts
window.initVideoJs = import("./videojs-imports.ts")

// in base-theme/layouts/partials/youtube_player.html
initVideoJs()
```

I think there are two advantages to the dynamic import:
1. webpack will automatically split `course-v2/assets/videojs-imports.ts` into a separate chunk out for us (I'm pretty sure)
2. `import()`ed code should run after first render, so I think the page will be interactive faster

Webpack compiles `import()` statements to its own version (not ES6 imports), but they still use promises. So the "downside" would be that it wouldn't work on IE11. But apparently YouTube doesn't support IE11, and our videos are  YouTube embeds, so that's probably OK.

_Re browser support: We do not have a hard rule. The goal is to generally work on old browsers, and provide full functionality on newer browsers (say, anything beyond IE11). _